esp/services/ws_store/espstorelib/daliKVStore.cpp

```diff
@@ -57,6 +58,15 @@ bool CDALIKVStore::createStore(const char * apptype, const char * storename, con
     if (maxvalsize != 0)
         apptree->setPropInt(DALI_KVSTORE_MAXVALSIZE_ATT, maxvalsize);
 
+    conn->changeMode(RTM_LOCK_WRITE, SDS_LOCK_TIMEOUT_KVSTORE);
```

Comment:
I realized there's a potential problem with this approach.
If 1 client is at this point, then this changeMode will succeed, however if >1 client is here, then they all hold a read lock, and no client will be able to obtain an [exclusive] write lock because others will still have read locks.

The implementation will need to relinquish the read lock before attempting to get the exclusive lock, e.g.:
```
    // clear read lock 1st, then establish a write lock and recheck
    conn.clear();
    conn.setown(querySDS().connect(DALI_KVSTORE_PATH, myProcessSession(), RTM_LOCK_WRITE, SDS_LOCK_TIMEOUT_KVSTORE));
```

Reply:
minor: I'd probably perform this rechecking before the 'apptree' creation code, i.e. immediately after line 39.